docs/developer-guide/pprofiling.md

```diff
@@ -0,0 +1,146 @@
+# Performance Profiling in zot
+
+> :point_right: Use zot's built-in profiling tools to collect and analyze runtime performance.
+
+The profiling capabilities within zot allow a zot [administrator](../articles/authn-authz.md) to collect and export a range of diagnostic performance data such as CPU intensive function calls, memory allocations, and execution traces. The collected data can then be analyzed using go tools and a variety of available visualization tools.
+
+## What data is available?
+
+The zot source code incorporates runtime analysis tools to collect data for the following performance-related profiles:
+
+| Profile  | Description |
+| ------- | --------- |
+| allocs | A sampling of all past memory allocations. |
+| block | Stack traces that led to blocking on synchronization primitives. |
+| cmdline | The command line invocation of the current program. |
+| goroutine| Stack traces of all current goroutines. Use `debug=2` as a URL query parameter to export in the same format as an unrecovered panic. |
+| heap | A sampling of memory allocations of live objects. You can specify the `gc` GET parameter to run GC before taking the heap sample. |
+| mutex | Stack traces of holders of contended mutexes. |
+| profile | CPU usage profile. You can specify the duration in the `seconds` URL query parameter. After receiving the profile file, use the `go tool pprof` command to investigate the profile. |
+| threadcreate | Stack traces that led to the creation of new OS threads. |
+| trace | A trace of execution of the current program. You can specify the duration in the `seconds` URL query parameter. After you get the trace file, use the `go tool trace` command to investigate the trace. |
+
+To return an HTML-format profile list along with a count of currently available records for each profile, use the following API command: 
+
+    /v2/_zot/pprof/
+
+> :pencil2: If authentication is enabled, only an admin user can access this API.
+
```

Comment:
## Supported Profile Types

`zot` incorporates [golang`s pprof package](https://pkg.go.dev/runtime/pprof) and therefore the following profile types are supported.

profile
heap
...